instrumentation/opentelemetry-instrumentation-cherrypy/src/opentelemetry/instrumentation/cherrypy/__init__.py

```diff
@@ -0,0 +1,140 @@
+        self.response_hook = kwargs.pop('response_hook', None)
+        excluded_urls = kwargs.pop('excluded_urls', None)
+        self._otel_excluded_urls = (_excluded_urls_from_env if excluded_urls is None else parse_excluded_urls(excluded_urls))
+        self._is_instrumented_by_opentelemetry = True
+        super().__init__(*args, **kwargs)
+        
+    def __call__(self, environ, start_response):
+        if self._otel_excluded_urls.url_disabled(environ.get('PATH_INFO', '/')):
+            return super().__call__(environ, start_response)
+
+        if not self._is_instrumented_by_opentelemetry:
+            return super().__call__(environ, start_response)
+        
+        start_time = time_ns()
+        span, token = _start_internal_or_server_span(
+            tracer=self._otel_tracer,
+            span_name=otel_wsgi.get_default_span_name(environ),
+            start_time=start_time,
+            context_carrier=environ,
+            context_getter=otel_wsgi.wsgi_getter,
+        )
+        if self.request_hook:
+            self.request_hook(span, environ)
+        attributes = otel_wsgi.collect_request_attributes(environ)
+        active_requests_count_attrs = (
+            otel_wsgi._parse_active_request_count_attrs(attributes)
+        )
+        duration_attrs = otel_wsgi._parse_duration_attrs(attributes)
+        self.active_requests_counter.add(1, active_requests_count_attrs)
+
+        if span.is_recording():
+            for key, value in attributes.items():
+                span.set_attribute(key, value)
+            if span.is_recording() and span.kind == trace.SpanKind.SERVER:
+                custom_attributes = (
+                    otel_wsgi.collect_custom_request_headers_attributes(environ)
+                )
+                if len(custom_attributes) > 0:
+                    span.set_attributes(custom_attributes)
+
+        activation = trace.use_span(span, end_on_exit=True)
+        activation.__enter__()
+
+        def _start_response(status, response_headers, *args, **kwargs):
+            propagator = get_global_response_propagator()
+            if propagator:
+                propagator.inject(response_headers, setter=otel_wsgi.default_response_propagation_setter)
+            
+            if span:
+                otel_wsgi.add_response_attributes(span, status, response_headers)
+                status_code = otel_wsgi._parse_status_code(status)
+                if status_code is not None:
+                    duration_attrs[SpanAttributes.HTTP_STATUS_CODE] = status_code
+                if span.is_recording() and span.kind == trace.SpanKind.SERVER:
+                    custom_attributes = otel_wsgi.collect_custom_response_headers_attributes(response_headers)
+                    if len(custom_attributes) > 0:
+                        span.set_attributes(custom_attributes)
+            
+            if self.response_hook:
+                self.response_hook(span, status, response_headers)
```

Comment:
Please add tests. If the PR is not ready then please mark it a draft.

Reply:
Hey, I'm really sorry. I forgot to mark this PR draft. There's still lot of work required in this PR.